src/mappings/mappingHandlers.ts

```diff
@@ -399,13 +462,89 @@ export async function handleNativeBalanceDecrement(event: CosmosEvent): Promise<
     const coin = parseCoins(amountStr)[0];
     const amount = BigInt(0) - BigInt(coin.amount); // save a negative amount for a "spend" event
     spendEvents.push({spender: spender, amount: amount, denom: coin.denom})
-  };
+  }
 
   for (const [i, spendEvent] of Object.entries(spendEvents)) {
     await saveNativeBalanceEvent(`${messageId(event)}-spend-${i}`, spendEvent.spender, spendEvent.amount, spendEvent.denom, event);
   }
 }
 
+export async function handleContractStoreEvent(event: CosmosEvent): Promise<void> {
+  logger.info(`[handleContractStoreEvent] (tx ${event.msg.tx.hash}): indexing event ${messageId(event.msg)}`)
+  logger.debug(`[handleContractStoreEvent] (event.event): ${JSON.stringify(event.event, null, 2)}`)
+  logger.debug(`[handleContractStoreEvent] (event.log): ${JSON.stringify(event.log, null, 2)}`)
+
+  const id = messageId(event.msg);
+  let codeId;
+  const sender = event.msg?.msg?.decodedMsg?.sender;
+  const permission = event.msg?.msg?.decodedMsg?.permission;
+
+  for (const [_, e] of Object.entries(event.event.attributes)) {
+    if (e.key === "code_id") {
+      codeId = e.value;
+    }
+  }
```

Comment:
You can simplify to something like this:

```suggestion
  const code_attr = event.event.attributes.find((e) => e.key === "code_id");
  const code_id = code_attr?.value;
```